### PR TITLE
add visualizer plugin for new frontend system

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

As seen at https://youtu.be/ONMBYnhxnNU?t=883

I marked it as private for now but if we're happy with `@backstage/plugin-visualizer` we could publish it straight away. Another contender is to add this as an alpha export to `@backstage/plugin-devtools`, although I think it could also fit as a plugin that adds content to the devtools page.

Graph view:

image

Detailed view:

image

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by freben** [comment]
Remember to DCO sign your commits 🙏

File: plugins/visualizer/src/plugin.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+export const visualizerNavItem = createNavItemExtension({
+  title: 'Visualizer',
+  icon: VisualizerIcon,
+  routeRef: rootRouteRef,
+});
+
+/** @public */
+export const visualizerPlugin = createPlugin({
+  id: 'visualizer',
```
Comment (freben) [inline plugins/visualizer/src/plugin.tsx:43]: may perhaps consider naming both this and the package something muck less generic, like `app-structure-visualizer` or so; "visualizer" in a Backstage context is almost a catch-all :P

Comment (freben) [reply plugins/visualizer/src/plugin.tsx:43]: Another thought - does all of this actually fit within the existing `devtools` package instead?

**Review by vinzscam — APPROVED** [review]
nice!

**Comment by Rugvip** [comment]
Shipping this as a private package for now

**Pull request merged**